code/modules/projectiles/firing.dm

```diff
@@ -47,7 +50,26 @@
 			BB.p_x = text2num(mouse_control["icon-x"])
 		if(mouse_control["icon-y"])
 			BB.p_y = text2num(mouse_control["icon-y"])
+		if(mouse_control["screen-loc"])
+			//Split screen-loc up into X+Pixel_X and Y+Pixel_Y
+			var/list/screen_loc_params = text2list(mouse_control["screen-loc"], ",")
 
+			//Split X+Pixel_X up into list(X, Pixel_X)
+			var/list/screen_loc_X = text2list(screen_loc_params[1],":")
+
+			//Split Y+Pixel_Y up into list(Y, Pixel_Y)
+			var/list/screen_loc_Y = text2list(screen_loc_params[2],":")
+			// world << "X: [screen_loc_X[1]] PixelX: [screen_loc_X[2]] / Y: [screen_loc_Y[1]] PixelY: [screen_loc_Y[2]]"
+			var/x = text2num(screen_loc_X[1]) * 32 + text2num(screen_loc_X[2]) - 32
+			var/y = text2num(screen_loc_Y[1]) * 32 + text2num(screen_loc_Y[2]) - 32
+			var/ox = round(480/2) //"origin" x - Basically center of the screen. This is a bad way of doing it because if you are able to view MORE than 15 tiles at a time your aim will get fucked.

```

Comment:
porting the screen size calculation code fox linked is pretty easy, i did it in my tests to see if it would fix the aiming problems i was having in those .gifs. it doesn't, but it would fix the issue remie is talking about.